hypertools/tools/cluster.py

```diff
@@ -1,11 +1,15 @@
 #!/usr/bin/env python
 
 from sklearn.cluster import KMeans
-from sklearn.decomposition import PCA
+import warnings
 import numpy as np
 from .._shared.helpers import *
+from .reduce import reduce as reducer
+from .align import align as aligner
+from .normalize import normalize as normalizer
 
-def cluster(x, n_clusters=8, ndims=None):
+def cluster(x, n_clusters=8, ndims=None, model='IncrementalPCA',
```

Comment:
the `reduce_params` syntax (and `cluster_params` and `align_params`) looks good.  that also brings up a good point-- the example i posted won't work, since it doesn't specify the number of k-means clusters.  so it should actually be something like:
`y = hyp.plot(x, reduce='TSNE', cluster='kmeans', cluster_params{'n': 6}, align='SRM', ndims=5)`

Re: keeping `model` and `model_params`, that approach (outputting a warning and depreciating them in the future) sounds good to me.